src/utils/Timer.h

```diff
@@ -46,15 +46,34 @@ class Timer {
           std::chrono::seconds totalDuration);
     ~Timer();
 
+    /**
+     * Start the timer thread setting the duration to the default valued defined in the constructor
+     */
     void start();
 
+    /**
+     * Stop the timer and join the timer thread
+     */
     void stop();
 
+    /**
+     * Get the actual state of the timer
+     * @return the timer state value
+     */
     TimerState getState() const { return timerState; }
 
+    /**
+     * Get the remaining duration of the timer
+     * @return the remaining time
+     */
```

Comment:
We should specify that this is only valid if `pause` is called.
_p.s. `pause` is not yet implemented_